inspire_disambiguation/core/ml/models.py

```diff
@@ -672,6 +675,26 @@ def fit(self, n_jobs=8):
         )
         self.clusterer.fit(self.X, self.y)
 
+    def score(self, test_uuids, labels):
+        """
+        Return the clustering statistics (b3 precision, b3 recall, b3 f1 score)
+        and wrongly clustered samples for training, test and the whole dataset.
+
+        Args:
+            test_uuids - list of signatures uuids used for testing
+            labels - list of labels (author id) for test dataset
+        """
+        all_uuids = np.vectorize(lambda x: x.signature_uuid)(self.X).flatten()
+        test_uuids_array = np.array(test_uuids)
+        mask = np.array([item in test_uuids_array for item in all_uuids])
+        y_tr = self.y[~mask]
+        y_test = np.array(labels)
+        labels_tr = self.clusterer.labels_[~mask]
+        labels_val = self.clusterer.labels_[mask]
+        return compute_clustering_statistics(self.X, self.y, self.clusterer.labels_), \
+               b3_precision_recall_fscore(y_tr, labels_tr), \
+               b3_precision_recall_fscore(y_test, labels_val)
```

Comment:
all these functions return precision, recall and f-score (and the first one the wrong labels on top), but when calling it you assume you just receive the f-score.